carbon-design-system/carbon-addons-iot-react
# fix(datetimepickerv2): remove default time value from humanValue object

Closes #3631 

**Summary**

- The DateTimePicker and DateTimePickerV2 always auto populates the default value 00:00 if time is not select. Now we set time value default to null. It would show empty on TimePickerDropDown in DateTimePickerV2 and TimePickerSpinner. In the top input it would only show date if date is selected.


**Change List (commits, features, bugs, etc)**

- changed startTime and endTime default value to null for DateTimePicker, DateTimePickerV2WithNewTimeSpinner, DateTimePickerV2WithoutNewTimeSpinner
- updated component tests and e2e tests

**Acceptance Test (how to verify the PR)**

- As in captured video

https://user-images.githubusercontent.com/24279794/204655032-eb7b349b-60c0-44e8-a035-3d82d2863849.mov

Single select Test
- go to single select [story](https://deploy-preview-3656--carbon-addons-iot-react.netlify.app/?path=/story/2-watson-iot-experimental-%E2%98%A2%EF%B8%8F-datetimepickerv2--single-select)
- open dateTimePicker menu
- click on clear button
- open dateTimePicker menu again
- select a date
- check the value in the input does not have default 00:00

Range select test
- go to Default [story](https://deploy-preview-3656--carbon-addons-iot-react.netlify.app/?path=/story/2-watson-iot-experimental-%E2%98%A2%EF%B8%8F-datetimepickerv2--default)
- check useNewTimeSpinner checkbox in the knobs
- open dateTimePIcker preset menu
- click on custom range
- click on absolute radio button
- select a start date
- check there is no default time 00:00 show up in the input
- select an end date
- - check there is no default time 00:00 show up in the input

Same for Range select without check useNewTimeSpinner in the Default [story](https://deploy-preview-3656--carbon-addons-iot-react.netlify.app/?path=/story/2-watson-iot-experimental-%E2%98%A2%EF%B8%8F-datetimepickerv2--default)

Same for Date Time Range Selector/ Default [story](https://deploy-preview-3656--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-date-time-range-selector--default)


**Regression Test (how to make sure this PR doesn't break old functionality)**

- tests here



**Things to look for during review**
